warehouse/query-core/src/main/java/datawave/query/index/lookup/RangeStream.java

```diff
@@ -475,22 +506,30 @@ public ScannerStream visit(ASTEQNode node, Object data) {
         }
 
         // Check if field is not indexed
+        linesPrinted++;
         if (!isIndexed(fieldName, config.getIndexedFields())) {
             try {
                 if (this.getAllFieldsFromHelper().contains(fieldName)) {
-                    log.debug("{\"" + fieldName + "\": \"" + literal + "\"} is not indexed.");
+                    if (maxLinesToPrint > 0 && linesPrinted > maxLinesToPrint) {
+                        log.debug("{\"" + fieldName + "\": \"" + literal + "\"} is not indexed.");
+                    }
```

Comment:
Likewise it could be useful to add a single log message that states we exceeded the threshold and won't print any more, but that's less straightforward to implement here. Alternately, logging at trace level after we hit the threshold could be an alternate approach because if we _do_ need the full log lines for some reason on an operational system, we can dynamically change the log level. Otherwise the only alternative would be to patch the configuration, restart and re-run the query.

Reply:
I like flipping to trace logging if the threshold is exceeded